src/cloudinaryUploader.js

```diff
@@ -74,10 +74,41 @@ const DEFAULT_REQUEST_OPTIONS = {
   mode: 'cors',
 }
 
-// The default public id creator returns the sanitized name of the file via html escaping.
-// Source: https://support.cloudinary.com/hc/en-us/articles/115001317409--Legal-naming-conventions
+// Test for all forbidden characters (defined by Cloudinary) and spaces
+const FILE_NAME_PATTERN = /[\s?&#\\%<>]/gi
+
+/**
+ * The default public id creator returns the sanitized name of the file by
+ * replacing invalid characters (incl. those that may be html escaped).
+ * Cloudinary does _not_ handle html escaping well via the API our their
+ * dashboard so it's best to avoid it entirely.
+ * 
+ * Source: https://support.cloudinary.com/hc/en-us/articles/115001317409--Legal-naming-conventions
+ */
 function defaultCreatePublicId (file) {
-  return encodeURIComponent(file.name)
+  const decodedFileName = decodeFileName(file.name || 'file_upload_' + Date.now())
```

Comment:
I'd assign this to a variable, this line is a lil unwieldy 😀

Reply:
Fair enough 😅 . I changed it from a template literal just for you!